datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -2763,6 +2789,52 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_array_replace_with_null() {
+        // ([3, 1, NULL, 3], 3,    4,    2)  => [4, 1, NULL, 4] NULL not matched
+        // ([3, 1, NULL, 3], NULL, 5,    2)  => [3, 1, NULL, 3]  NULL  not replaced (not eq)
```

Comment:
why null is not replaced with 5 => [3, 1, 5, 3]?

Reply:
It seems that arrow_ord::cmp::eq([3,1,null,3], null) return [null,null,null,null] which I expect to see [false,false,true,false]

Reply:
I believe it's reasonable for arrow-rs to handle null comparisons

Reply:
I think it is because in SQL, `NULL = NULL` is *NOT* `true` like you might expect, but is actually `NULL`

If we want the `NULL = NULL` = true behavior, we could use the `not_distinct` kernel instead: https://docs.rs/arrow/latest/arrow/compute/kernels/cmp/fn.not_distinct.html

Reply:
> I think it is because in SQL, `NULL = NULL` is _NOT_ `true` like you might expect, but is actually `NULL`
> 
> If we want the `NULL = NULL` = true behavior, we could use the `not_distinct` kernel instead: https://docs.rs/arrow/latest/arrow/compute/kernels/cmp/fn.not_distinct.html

I would prefer having `not_distinct` here, also there is no other DB have the similar function https://github.com/apache/arrow-datafusion/issues/7074, so I think we can have our own definition of what `array_replace` do.

Reply:
ok, sounds good. I will update this PR

Reply:
Done in acbffa0e219541eef168d205dd773e1864d0dfb7